From dc6db28b9395a90ef80ba7c7997d80214ee17a5c Mon Sep 17 00:00:00 2001
Subject: [PATCH] susfs: fix fsnotify compatibility for kernel 4.19

Fix compilation errors in fs/susfs.c due to fsnotify infrastructure
differences between newer kernels and version 4.19.

- Update susfs_handle_sdcard_inode_event signature to match 4.19 expectations
  by removing inode_mark and vfsmount_mark parameters.
- Fix fsnotify_add_mark call to use inode connector and FSNOTIFY_OBJ_TYPE_INODE.

---
 fs/susfs.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/fs/susfs.c b/fs/susfs.c
--- a/fs/susfs.c
+++ b/fs/susfs.c
@@ -1126,8 +1126,6 @@ static int watch_one_dir(struct watch_dir *wd)
  */
 static int susfs_handle_sdcard_inode_event(struct fsnotify_group *group,
 											struct inode *to_tell,
-											struct fsnotify_mark *inode_mark,
-											struct fsnotify_mark *vfsmount_mark,
 											u32 mask, const void *data, int data_type,
 											const unsigned char *file_name, u32 cookie,
 											struct fsnotify_iter_info *iter_info)
@@ -1160,7 +1158,7 @@ static int add_mark_on_inode(struct inode *inode, u32 mask,
 	fsnotify_init_mark(m, g);
 	m->mask = mask;
 
-	if (fsnotify_add_mark(m, inode, NULL, 0)) {
+	if (fsnotify_add_mark(m, &inode->i_fsnotify_marks, FSNOTIFY_OBJ_TYPE_INODE, 0)) {
 		fsnotify_put_mark(m);
 		return -EINVAL;
 	}